tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
```

Comment:
On 2015-02-17 06:22, Randall Leeds wrote:
> I /definitely/ do not like that |scrollIntoView| is written the same
> as the proposed standard but does not take the same arguments.

Returning to this a bit.

The challenge here is that the library we are currently using for
scrolling uses a completely different set of options
<https://github.com/hypothesis/h/blob/2fa9382fea059ac8fa8dbccce103959cd74c519f/h/static/scripts/vendor/jquery.scrollintoview.js#L74>,
compared to those defined by the proposed standard
<http://dev.w3.org/csswg/cssom-view/#dom-element-scrollintoview>.

To make progress with this, we could evaluate the various available
polyfills / libraries that provide a similar scrolling functionality
.... if we really wanted to, but I don't feel that this a priority for
us now.

On the other hand, to avoid the clashing names, we could also rename our
method in order to avoid the impression that we support the standard.
(For example, we could call it scrollToView ).